src/edu/ucsb/nceas/metacat/util/SystemUtil.java

```diff
@@ -52,669 +52,666 @@
-	 * file separator + the context
-	 * 
-	 * @return string holding the server URL with context
-	 */
-	public static String getContextURL() throws PropertyNotFoundException {
-		return getServerURL() + "/"
-				+ PropertyService.getProperty("application.context");
-	}
-	
-	/**
-	 * Get the secure server URL with the context. This is made up of the secure server URL +
-	 * file separator + the context
-	 * 
-	 * @return string holding the server URL with context
-	 */
-	public static String getSecureContextURL() throws PropertyNotFoundException {
-		return getSecureServerURL() + "/"
-				+ PropertyService.getProperty("application.context");
-	}
-
-	/**
-	 * Get the servlet URL. This is made up of the server URL with context +
-	 * file separator + the metacat servlet name
-	 * 
-	 * @return string holding the servlet URL
-	 */
-	public static String getServletURL() throws PropertyNotFoundException {
-		return getContextURL() + "/" + METACAT_SERVLET;
-	}
-	
-	/**
-	 * Get the internal context url. If the internal server url is not accessible, it falls back to the external context url.
-	 * @return  the url of the internal Metacat context
-	 * @throws PropertyNotFoundException
-	 */
-	public static String getInternalContextURL() throws PropertyNotFoundException {
-	    return getInternalServerURL() + "/"
+    private static Log logMetacat = LogFactory.getLog(SystemUtil.class);
+    private static String METACAT_SERVLET = "metacat";
+//    private static String METACAT_WEB_SERVLET = "metacatweb";
+    private static int OS_CLASS = 0;
+    private static boolean firstTimeTryInternalURL = true;
+    private static boolean firstTryInternalURLAfterFullInit = true;
+    private static String internalURL = null;
+    private static boolean internalURLReplacedByExternal = false;
+    
+    // Class of OS.  If we need more granularity, we should create a version
+    // list and access it separately.
+    public static int WIN_OS = 1;
+    public static int LINUX_OS = 2;
+    public static int MAC_OS = 3;
+    public static int OTHER_OS = 4;
+    
+    /**
+     * private constructor - all methods are static so there is no no need to
+     * instantiate.
+     */
+    private SystemUtil() {}
+
+    /**
+     * Get the OS for this system.
+     * @return an integer representing the class of OS.  Possibilities are:
+     *     WIN_OS = 1;
+     *     LINUX_OS = 2;
+     *     MAC_OS = 3;
+     *     OTHER_OS = 4;
+     */
+    public static int getOsClass() {
+        if (OS_CLASS > 0) {
+            return OS_CLASS;
+        }
+        
+        String osName = System.getProperty("os.name");
+        if (osName.startsWith("Windows")) {
+            OS_CLASS =  WIN_OS;
+        } else if (osName.startsWith("Linux")) {
+            OS_CLASS =  LINUX_OS;
+        } else if (osName.startsWith("Mac")) {
+            OS_CLASS =  MAC_OS;
+        } else if (osName.startsWith("Mac")) {
+            OS_CLASS =  OTHER_OS;
```

Comment:
I know you didn't change this, but I noticed these two `else` conditions are exactly the same. Should the last one be like this instead?:
```java
} else {
    OS_CLASS =  OTHER_OS;
}
```